site/content/docs/howto/configure-supervisor-with-auth0.md

```diff
@@ -0,0 +1,141 @@
+title: Configure the Pinniped Supervisor to use Auth0 as an OIDC provider
+description: Set up the Pinniped Supervisor to use Auth0 login.
+cascade:
+  layout: docs
+menu:
+  docs:
+    name: Configure Supervisor With Auth0 OIDC
+    weight: 80
+    parent: howtos
+---
+The Supervisor is an [OpenID Connect (OIDC)](https://openid.net/connect/) issuer that supports connecting a single
+"upstream" identity provider to many "downstream" cluster clients.
+
+This guide shows you how to configure the Supervisor so that users can authenticate to their Kubernetes
+cluster using their Auth0 credentials.
+
+## Prerequisites
+
+This how-to guide assumes that you have already [installed the Pinniped Supervisor]({{< ref "install-supervisor" >}}) with working ingress,
+and that you have [configured a FederationDomain to issue tokens for your downstream clusters]({{< ref "configure-supervisor" >}}).
+
+## Create an Auth0 Application
+
+Follow the instructions to [create an application](https://auth0.com/docs/get-started/auth0-overview/create-applications).
+
+For example, to create an app:
+
+1. In the [Auth0 Admin Console](https://manage.auth0.com/), navigate to _Applications_ > _Applications_.
+2. Create a new application:
+   1. Click `+ Create Application`.
+   2. Provide a name. For `Choose an application type`, select `Regular Web Applications`.
+   3. Under `Settings`:
+      1. Note the `Client ID` and `Client Secret`, which will be provided later to Pinniped.
+      2. Update `Allowed Callback URLs` with Pinniped's issuer URL, appending `/callback` to the end. (Example: `https://pinniped.issuer.example.com/callback).
+      3. Under `Advanced Settings`, choose `Grant Types` and make sure that `Authorization Code`, `Refresh Token`, and `Password` are selected.
+      4. Also under `Advanced Settings`, find the Auth0 Issuer URL by loading the URL at `Endpoints` > `OAuth` > `OpenID Configuration` and finding the `"issuer"` field.
+
+## Configure Auth0 to include user groups in its ID tokens
+
+There are a handful of ways to include user groups into the Auth0 ID token so that Kubernetes RBAC can be used to assign permissions on the cluster.
+
+**NOTE**: Pinniped does not have a specific recommendation for how user groups are defined in Auth0.
+The examples below are provided as examples to better understand how Auth0 and Pinniped integrate, not how to configure Auth0.
+
+One way to add user groups into the ID token is to add a [custom claim](https://auth0.com/blog/adding-custom-claims-to-id-token-with-auth0-actions/).
+Auth0 recommends using a [namespaced format](https://auth0.com/docs/secure/tokens/json-web-tokens/create-custom-claims) for your claim names.
+Pinniped requires that the group claim is an array of strings.
+
+In this example, replace `"https://example.com/pinniped/groups"` with the namespaced claim name of your choice.
+The array of groups should be dynamically provisioned from the Auth0 user store, never hardcoded as shown in this example.
```

Comment:
How about something like this?

```suggestion
Auth0 does not have a simple concept of group membership for users. It may possible to model group membership in Auth0, but the specifics depend on which enterprise connector or database is used to create your users. Please refer to the Auth0 documentation for more information.

Pinniped does not have a specific recommendation for how user groups are defined in Auth0. The examples below are provided as examples to better understand how Auth0 and Pinniped integrate, not how to configure Auth0.

Assuming that you have somehow configured Auth0 to include group membership information about your users, you can expose this to Pinniped by configuring Auth0 to include a [custom claim](https://auth0.com/blog/adding-custom-claims-to-id-token-with-auth0-actions/) in the Auth0 ID token.

Auth0 recommends using a [namespaced format](https://auth0.com/docs/secure/tokens/json-web-tokens/create-custom-claims) for your custom claim names. In the following example, replace `"https://example.com/pinniped/groups"` with the namespaced claim name of your choice. Pinniped requires that the value of the group claim is an array of strings.

The following example is intended to show how to add a custom claim, but does not show a realistic example of where the group names should come from. To keep this example simple, the group names shown here are hardcoded. Do not hardcode group names for a production system. Instead, the array of groups should be dynamically provisioned from the appropriate place in the Auth0 user store.
```

Reply:
done with next push